src/app/note-list/note-list.component.ts

```diff
@@ -0,0 +1,20 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Note } from '../models/note';
+
+@Component({
+  selector: 'app-note-list',
+  templateUrl: './note-list.component.html',
+  styleUrls: ['./note-list.component.scss']
+})
+export class NoteListComponent {
+  @Input() noteList!: Note[];
+  @Input() displayedColumns!: string[];
+  @Output() addNote = new EventEmitter();
+  @Output() editNote: EventEmitter<Note> = new EventEmitter<Note>();
+  @Output() deleteNote: EventEmitter<number> = new EventEmitter<number>();
+
+  onClick(event: any, id: number) {
+    event.stopPropagation();
```

Comment:
This can all be done in the template. `(click)="$event.stopPropagation(); deleteNote.emit(id);"`